File: app/src/main/java/com/service/saver/saverservice/tumblr/TumblrActivity.java
package com.service.saver.saverservice.tumblr;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.service.saver.saverservice.MyApp;
import com.service.saver.saverservice.R;
import com.service.saver.saverservice.tumblr.util.JumblrHolder;
import com.tumblr.jumblr.types.Post;

import java.net.URL;
import java.util.List;

import needle.Needle;

public class TumblrActivity extends AppCompatActivity {
    public static JumblrHolder client = new JumblrHolder();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_tumblr);
        Bundle extras = getIntent().getExtras();
        try {
            String text = (String) getIntent().getExtras().get(Intent.EXTRA_TEXT);
            if (text.contains("tumblr")) {
                URL url = new URL(text);
                String host = url.getHost();
                String path = url.getPath();
                String[] hostsplit = host.split("\\.");
                String[] pathsplit = path.split("/");
                String id = pathsplit[2];
                String blog = hostsplit[0];
                Needle.onBackgroundThread().execute(() -> {
                    if (!(client.TOEKN_SECRET.isEmpty() && client.TOKEN_KEY.isEmpty())) {
                        Post post = client.blogPost(blog, Long.valueOf(id));
                        List<String> urlFile = JumblrHolder.getUrlFile(post);
                        urlFile.forEach(MyApp::add);
                    }
                });

            }
            finish();

        } catch (Exception e) {
            e.printStackTrace();
        }

    }
}

File: app/src/main/java/com/service/saver/saverservice/tumblr/TumblrActivity.kt
package com.service.saver.saverservice.tumblr

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.service.saver.saverservice.MainTabActivity.client
import com.service.saver.saverservice.R
import com.service.saver.saverservice.domain.PostLink
import com.service.saver.saverservice.domain.UserLink
import com.service.saver.saverservice.sqllite.AdminSQLiteOpenHelper
import com.service.saver.saverservice.tumblr.util.TumblrClient
import needle.Needle
import java.net.URL

class TumblrActivity : AppCompatActivity() {
    private var db: AdminSQLiteOpenHelper? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_tumblr)
        db = AdminSQLiteOpenHelper(this)
        try {
            val text = intent.extras!!.get(Intent.EXTRA_TEXT) as String
            if (text.contains("tumblr")) {
                val url = URL(text)
                val host = url.host
                val path = url.path
                val hostsplit = host.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                val pathsplit = path.split("/".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                val id = pathsplit[2]
                val blog = hostsplit[0]
                Needle.onBackgroundThread().execute {
                    if (!(TumblrClient.TOKEN_SECRET!!.isEmpty() && TumblrClient.TOKEN_KEY!!.isEmpty())) {
                        val post = client!!.blogPost(blog, java.lang.Long.valueOf(id))
                        val urlFile = TumblrClient.getUrlFile(post)
                        urlFile.forEach {
                            var postlink = PostLink()
                            postlink.url = it
                            postlink.username = post.blogName
                            val postLink = this.db!!.getPostLink(it)
                            if (postLink == null) {
                                this.db!!.agregarPostLink(postlink)
                            }
                        }
                    }
                }

            }

            if (text.contains("twitter") and text.contains("@")) {
                var user = text.split("@")[1].split(")")[0];
                var userLink = UserLink();
                userLink.username = user;
                val findUserLink = this.db!!.getUserLink(user)
                if (findUserLink == null) {
                    this.db!!.agregarUserLink(userLink)
                }
            }
            finish()

        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

}